src/main/java/org/gridsuite/securityanalysis/server/repositories/specifications/SpecificationUtils.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.securityanalysis.server.repositories.specifications;
+
+import jakarta.persistence.criteria.Path;
+import jakarta.persistence.criteria.Root;
+import org.gridsuite.securityanalysis.server.dto.ResourceFilterDTO;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.query.EscapeCharacter;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+import static org.springframework.data.jpa.domain.Specification.anyOf;
+import static org.springframework.data.jpa.domain.Specification.not;
+
+/**
+ * Utility class to create Spring Data JPA Specification (Spring interface for JPA Criteria API).
+ *
+ * @author Kevin Le Saulnier <kevin.lesaulnier@rte-france.com>
+ */
+public final class SpecificationUtils {
+
+    public static final String FIELD_SEPARATOR = ".";
+
+    // Utility class, so no constructor
+    private SpecificationUtils() {
+    }
+
+    // we use .as(String.class) to be able to works on enum fields
+    public static <X> Specification<X> equals(String field, String value) {
+        return (root, cq, cb) -> cb.equal(cb.upper(getColumnPath(root, field)).as(String.class), value.toUpperCase());
+    }
+
+    public static <X> Specification<X> contains(String field, String value) {
+        return (root, cq, cb) -> cb.like(cb.upper(getColumnPath(root, field)), "%" + EscapeCharacter.DEFAULT.escape(value).toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
+    }
+
+    public static <X> Specification<X> startsWith(String field, String value) {
+        return (root, cq, cb) -> cb.like(cb.upper(getColumnPath(root, field)), EscapeCharacter.DEFAULT.escape(value).toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
+    }
+
+    public static <X> Specification<X> notEqual(String field, Double value) {
+        return (root, cq, cb) -> cb.notEqual(getColumnPath(root, field), value);
+    }
+
+    public static <X> Specification<X> lessThanOrEqual(String field, Double value) {
+        return (root, cq, cb) -> cb.lessThanOrEqualTo(getColumnPath(root, field), value);
+    }
+
+    public static <X> Specification<X> greaterThanOrEqual(String field, Double value) {
+        return (root, cq, cb) -> cb.greaterThanOrEqualTo(getColumnPath(root, field), value);
+    }
+
+    public static <X> Specification<X> isNotEmpty(String field) {
+        return (root, cq, cb) -> cb.isNotEmpty(getColumnPath(root, field));
+    }
+
+    public static <X> Specification<X> distinct() {
+        return (root, cq, cb) -> {
+            cq.distinct(true);
+            return null;
```

Comment:
Ok get it, we just create a `null` specification to set the distinct property

Reply:
Maybe add a small comment to explain why we create a null specification and that the distinct works because the criteria query propagate to the following specifications, if I understood well ?

Reply:
I added a comment, tell me if it's not clear enough

Reply:
Did you push it ? I don't see it